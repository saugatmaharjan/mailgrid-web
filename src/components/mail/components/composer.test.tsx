import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";

import Composer from "./composer";
import Theme from "components/theme";

afterEach(cleanup);

it("renders a composer", () => {
  const { asFragment } = render(
    <Theme>
      <Composer />
    </Theme>
  );
  expect(asFragment()).toMatchSnapshot();
});

describe("Recipient field", () => {
  it("field rendered in dom and is required", () => {
    const { getByTestId } = render(
      <Theme>
        <Composer />
      </Theme>
    );
    expect(getByTestId("to")).toBeInTheDocument();
    expect(getByTestId("to")).toBeRequired();
  });
});

describe("Subject field", () => {
  it("field rendered in dom and is not required", () => {
    const { getByTestId } = render(
      <Theme>
        <Composer />
      </Theme>
    );
    expect(getByTestId("to")).toBeInTheDocument();
    expect(getByTestId("subject")).not.toBeRequired();
  });
});

describe("Message field", () => {
  it("field rendered in dom and is not required", () => {
    const { getByTestId } = render(
      <Theme>
        <Composer />
      </Theme>
    );
    expect(getByTestId("message")).toBeInTheDocument();
    expect(getByTestId("subject")).not.toBeRequired();
  });
});

describe("CC field", () => {
  it("field rendered in dom and is not required", () => {
    const { getByTestId } = render(
      <Theme>
        <Composer />
      </Theme>
    );
    fireEvent.click(getByTestId("cc-toggle"));
    expect(getByTestId("cc")).toBeInTheDocument();
    expect(getByTestId("cc")).not.toBeRequired();
  });
});

describe("BCC field", () => {
  it("field rendered in dom and is not required", () => {
    const { getByTestId } = render(
      <Theme>
        <Composer />
      </Theme>
    );
    fireEvent.click(getByTestId("bcc-toggle"));
    expect(getByTestId("bcc")).toBeInTheDocument();
    expect(getByTestId("bcc")).not.toBeRequired();
  });
});

describe("Test send", () => {
  it("Fill up all the fields", async () => {
    const { getByTestId } = render(
      <Theme>
        <Composer />
      </Theme>
    );

    const reciever = "maharjansaugat@gmail.com";
    fireEvent.change(getByTestId("to"), { target: { value: reciever } });
    expect(getByTestId("to")).toHaveValue(reciever);

    const subject = "Testing Library";
    fireEvent.change(getByTestId("subject"), { target: { value: subject } });
    expect(getByTestId("subject")).toHaveValue(subject);

    const message = "testing from testing library";
    fireEvent.change(getByTestId("message"), { target: { value: message } });
    expect(getByTestId("message")).toHaveValue(message);

    fireEvent.click(getByTestId("submit-btn"));
    expect(getByTestId("submit-btn")).toBeDisabled();
  });
});
