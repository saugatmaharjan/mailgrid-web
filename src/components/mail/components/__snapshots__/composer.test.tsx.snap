// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a composer 1`] = `
<DocumentFragment>
  <div
    class="sc-AxiKw gxOFcj"
  >
    <div
      class="sc-AxhCb exIHPI"
    >
      <span
        class="text-light"
      >
        To
      </span>
      <input
        class="sc-AxhUy cxbfHR flex-one"
        data-testid="to"
        name="to"
        required=""
        type="text"
        value=""
      />
      <span
        class="sc-AxmLO hYTuik text-light"
        data-testid="cc-toggle"
      >
        CC
      </span>
      <span
        class="sc-AxmLO hYTuik text-light"
        data-testid="bcc-toggle"
      >
        BCC
      </span>
    </div>
    <div
      class="sc-AxhCb exIHPI"
    >
      <span
        class="text-light"
      >
        Subject
      </span>
      <input
        class="sc-AxhUy cxbfHR flex-one"
        data-testid="subject"
        name="subject"
        type="text"
        value=""
      />
    </div>
    <textarea
      class="sc-AxgMl iggwgr"
      data-testid="message"
      name="text"
    />
    <div
      class="sc-AxheI iClzrt"
    >
      <button
        class="sc-Axmtr cuQrQz"
        data-testid="submit-btn"
        disabled=""
      >
        Send
      </button>
    </div>
  </div>
</DocumentFragment>
`;
